omod/src/main/java/org/openmrs/module/coreapps/fragment/controller/visit/RetrospectiveVisitFragmentController.java

```diff
@@ -31,31 +31,30 @@ public Object create(@SpringBean("adtService") AdtService adtService,
                                        @RequestParam(value = "stopDate", required = false) Date stopDate,
                                        HttpServletRequest request, UiUtils ui) {
 
-        // if no stop date, set it to start date
-        if (stopDate == null) {
-            stopDate = startDate;
-        }
 
         // set the startDate time component to the start of day
         startDate = new DateTime(startDate).withTime(0,0,0,0).toDate();
 
         // if stopDate is today, set stopDate to current datetime, otherwise set time component to end of date
+         if (stopDate != null){
```

Comment:
Why enclose these in if stopDate != null?

Reply:
the methods enclosed here are for formatting the endDate which is not neccessary when the stopDate is null....